Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -2286,6 +2304,35 @@ repoTypeDirname Bazaar    = [".bzr"]
 repoTypeDirname Monotone  = ["_MTN"]
 repoTypeDirname Pijul     = [".pijul"]
 
+checkDesirableExtraDocFilesAreIncluded :: Monad m => CheckPackageContentOps m
+                                       -> PackageDescription
+                                       -> m [PackageCheck]
+checkDesirableExtraDocFilesAreIncluded ops pkg = do
+    let dir = "."
+    rootContents <- getDirectoryContents ops dir
+    desirable <- filterM (doesFileExist ops)
+                         [ dir </> file
+                         | file <- rootContents
+                         , isDesirableExtraDocFile file
+                         ]
+    -- [TODO] extraSrcFiles: add warning if the files are globed in
+    --        extra-source-files instead of extra-doc-files?
+    allDocFiles <- concatMap (fmap (dir </>)) <$>
+        traverse (getGlobFiles ops dir) (extraDocFiles pkg)
+    case desirable \\ allDocFiles of
+        []       -> return []
+        unlisted -> return
+            [PackageDistSuspiciousWarn (MissingExpectedExtraDocFiles unlisted)]
+
+isDesirableExtraDocFile :: FilePath -> Bool
+isDesirableExtraDocFile fp = map toLower basename `elem` desirable
+  where
+    (basename, _ext) = splitExtension fp
+    desirable =
+      [ "readme"
+      , "changelog"
```

Comment:
Is that all that Hackage looks for? E.g., no "changes" nor "history"?

Reply:
Good catch! I fixed it and added reference to `hackage-server` code.